src/main/java/tech/jhipster/lite/module/infrastructure/secondary/javadependency/gradle/GradleCommandHandler.java

```diff
@@ -249,7 +257,75 @@ public void handle(AddMavenPluginManagement command) {
 
   @Override
   public void handle(SetBuildProperty command) {
-    throw new NotImplementedException(NOT_YET_IMPLEMENTED);
+    Assert.notNull(COMMAND, command);
+
+    command
+      .buildProfile()
+      .ifPresent(buildProfile -> {
+        File scriptPlugin = scriptPluginForProfile(buildProfile);
+        if (!scriptPlugin.exists()) {
+          throw new MissingGradleProfileException(buildProfile);
+        }
+        addPropertyTo(command.property(), scriptPlugin);
+      });
+  }
+
+  private void addPropertyTo(BuildProperty property, File buildGradleFile) {
+    MandatoryReplacer replacer;
+    if (propertyExistsFrom(property.key(), buildGradleFile.toPath())) {
+      replacer = existingPropertyReplacer(property);
+    } else {
+      replacer = addNewPropertyReplacer(property);
+    }
+
+    fileReplacer.handle(
+      projectFolder,
+      ContentReplacers.of(
+        new MandatoryFileReplacer(
+          new JHipsterProjectFilePath(Path.of(projectFolder.folder()).relativize(buildGradleFile.toPath()).toString()),
+          replacer
+        )
+      )
+    );
+  }
+
+  private MandatoryReplacer existingPropertyReplacer(BuildProperty property) {
+    Pattern propertyLinePattern = Pattern.compile(
+      "val %s by extra\\(\"(.*?)\"\\)".formatted(toCamelCasedKotlinVariable(property.key())),
+      Pattern.MULTILINE
+    );
+    return new MandatoryReplacer(
+      new RegexReplacer(
+        (contentBeforeReplacement, replacement) -> propertyLinePattern.matcher(contentBeforeReplacement).find(),
+        propertyLinePattern
+      ),
+      convertToKotlinFormat(property)
+    );
+  }
+
+  private static MandatoryReplacer addNewPropertyReplacer(BuildProperty property) {
+    return new MandatoryReplacer(new RegexNeedleBeforeReplacer(always(), GRADLE_PROPERTY_NEEDLE), convertToKotlinFormat(property));
+  }
+
+  private static String convertToKotlinFormat(BuildProperty property) {
+    return "val %s by extra(\"%s\")".formatted(toCamelCasedKotlinVariable(property.key()), property.value().get());
+  }
+
+  private static String toCamelCasedKotlinVariable(PropertyKey key) {
+    return Arrays.stream(key.get().split("\\."))
+      .map(s -> s.substring(0, 1).toUpperCase() + s.substring(1).toLowerCase())
+      .collect(Collectors.collectingAndThen(Collectors.joining(), str -> str.substring(0, 1).toLowerCase() + str.substring(1)));
+  }
+
+  @ExcludeFromGeneratedCodeCoverage(reason = "The exception handling is hard to test and an implementation detail")
+  private boolean propertyExistsFrom(PropertyKey key, Path buildGradleProfileFile) {
+    try {
+      String content = Files.readString(buildGradleProfileFile);
+
+      return content.contains(toCamelCasedKotlinVariable(key));
```

Comment:
It might be more robust to verify the variable declaration, rather than just the variable appearance, no?
```suggestion
      return content.contains("val %s by extra(".formatted(toCamelCasedKotlinVariable(key));
```

Reply:
You are right! Only the key could find a name out of context.